exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/entities/CompositeIdTableEntityTest.kt

```diff
@@ -432,6 +432,35 @@ class CompositeIdTableEntityTest : DatabaseTestsBase() {
         }
     }
 
+    @Test
+    fun testIdParamWithCompositeValue() {
+        withTables(Towns) {
+            val townAValue = CompositeID {
+                it[Towns.zipCode] = "1A2 3B4"
+                it[Towns.name] = "Town A"
+            }
+            val townAId = Towns.insertAndGetId {
+                it[id] = townAValue
+                it[population] = 4
+            }
+
+            val query1 = Towns.selectAll().where { Towns.id eq idParam(townAId, Towns.id) }
+            val whereClause1 = query1.prepareSQL(this, prepared = true).substringAfter("WHERE ")
+            assertEquals("(${fullIdentity(Towns.zipCode)} = ?) AND (${fullIdentity(Towns.name)} = ?)", whereClause1)
+            assertEquals(4, query1.single()[Towns.population])
+
+            val query2 = Towns.selectAll().where { idParam(townAId, Towns.id).isNotNull() }
```

Comment:
To be honest, I don't finally understand what this statement means.

It looks like we want to query all the `Towns` that have non null id (that can not be null, because it's primary key). But even if we want to get all the entries with non null id (composite id), why do we need actual `townAId` value.

As I can see, the condition `where { idParam(townAId, Towns.id).isNotNull() }` generates the following sql `SELECT towns.zip_code, towns."name", towns.population FROM towns WHERE (towns.zip_code IS NOT NULL) AND (towns."name" IS NOT NULL)`

If I replace it with `where { Towns.id.isNotNull() }` I get the same SQL. 

I understand that it's the test for `idParam`, but this part may confuse in terms of how `idParam` is used.